server/src/auth-router.ts

```diff
@@ -74,41 +87,34 @@ if (DOMAIN) {
       domain: DOMAIN,
       clientID: CLIENT_ID,
       clientSecret: CLIENT_SECRET,
-      callbackURL:
-        (({
-          stage: 'https://commonvoice.allizom.org',
-          prod: 'https://commonvoice.mozilla.org',
-          dev: 'https://dev.voice.mozit.cloud',
-          sandbox: 'https://sandbox.voice.mozit.cloud',
-        } as any)[ENVIRONMENT] || '') + CALLBACK_URL,
-      scope: 'openid email',
+      callbackURL: (ENVIRONMENT_URLS[ENVIRONMENT] || '') + CALLBACK_URL,
     },
-    (
-      accessToken: any,
-      refreshToken: any,
-      extraParams: any,
-      profile: any,
-      done: any
-    ) => done(null, profile)
+    (profile: any, done: any) => done(null, profile)
   );
 
   passport.use(strategy);
 } else {
-  console.log('No Auth0 configuration found');
+  console.error('No Auth0 configuration found');
+}
+
+function parseState(request: Request) {
```

Comment:
The items that are on the `request.query` object are from the `qs` library and so could be an array or a string. To keep Typescript happy we have to check for that edgecase.